pmd-core/src/main/java/net/sourceforge/pmd/renderers/AbstractAccumulatingRenderer.java

```diff
@@ -80,9 +80,9 @@ public GlobalAnalysisListener newListener() throws IOException {
             this.start();
         }
 
-        return new GlobalAnalysisListener() {
-            final GlobalReportBuilderListener reportBuilder = new GlobalReportBuilderListener();
+        final GlobalReportBuilderListener reportBuilder = new GlobalReportBuilderListener();
```

Comment:
In my experience, if you put the final code before return, then there is no reportBuilder object available in the newly instantiated GlobalAnalysisListener and the variable cannot be accessed within the GlobalAnalysisListener and the compiler will report an error.
Please correct me if I have misunderstood.

Reply:
I think it has more to do with the scope of the variable than the final statement, If i understood you correctly?